### ISSUE TITLE
#[error(transparent)] triggers `clippy::used_underscore_binding` lint

### ISSUE THREAD

**Issue by U007D** [dtolnay/thiserror#88]
```rust 
#[derive(Debug, Error, PartialEq)]
pub enum Error {
    // ...
    #[error(transparent)]
    VersionError(#[from] SomeOtherVersionError),
}
```
yields
```
error: used binding `_0` which is prefixed with an underscore. A leading underscore signals that a binding will not be used.
 --> src/error.rs:9:13
  |
9 |     #[error(transparent)]
  |             ^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:7:5
  |
7 |     clippy::pedantic,
  |     ^^^^^^^^^^^^^^^^
  = note: `#[deny(clippy::used_underscore_binding)]` implied by `#[deny(clippy::pedantic)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#used_underscore_binding
```

**Comment by U007D** [comment]
<3

**Comment by dtolnay** [comment]
Fixed in 1.0.19.
